projects/self_feeding/self_feeding_agent.py

```diff
@@ -302,65 +302,83 @@ def act(self):
                 rating = self.extract_rating()
                 if rating == -1:
                     action = self.make_action(
-                        self.make_explanation_request(), reward=rating
+                        self.make_feedback_request(), reward=rating
                     )
-                    self.status = EXPLANATION_REQUESTED
+                    self.status = FEEDBACK_REQUESTED
 
                 else:
                     action, reply = self.make_rating_response(rating)
                     self.status = RATING_ACCEPTED
                     # Store just the non-template part of the response in the history
-                    self.history.append(reply)
+                    self.history.update_history({"text": reply})
 
-            elif self.status == EXPLANATION_REQUESTED:
+            elif self.status == FEEDBACK_REQUESTED:
                 action = self.make_action(THANKS + ' ' + NEWTOPIC)
                 self.status = NEWTOPIC_REQUESTED
-                self.history.clear()
+                self.history.reset()
 
             else:
                 raise Exception(f"Unrecognized status: {self.status}")
 
         if self.status == NORMAL:
             action = super().act()
-            self.history.append(action['text'])
+            self.history.update_history(action)
 
         return action
 
     def observe(self, observation):
         """Add to history, concatenate history-size utterances, and add person tokens"""
+
         # If their response is a response to a rating request, no work required
         if self.status == RATING_REQUESTED:
             self.last_rating = observation['text']
         else:
             if 'text' in observation:
-                self.history.append(observation['text'])
+                self.history.update_history(observation)
 
-        if len(self.history) > 0:
+        if len(self.history.history_strings) > 0:
+            # WARNING: Be suspicious of this; is history_size being applied correctly?
             observation['text'] = add_person_tokens(
```

Comment:
hmm. i don't think this is necessary. in any case we got rid of the function `add_person_tokens` in torch agent. instead, you should set the default arg `--add-person-tokens True`. then, the history will automatically be updated with this. is there a reason you did it this way?